diff --git a/src/handlers/mod.rs b/src/handlers/mod.rs
index 4fc2aea..65da055 100644
--- a/src/handlers/mod.rs
+++ b/src/handlers/mod.rs
@@ -15,13 +15,17 @@ use std::ffi::OsStr;
 use std::fs;
 use std::fs::{File, Metadata};
 use std::io::{self, Seek};
-use std::os::linux::fs::MetadataExt as _;
 use std::os::unix::fs as unix_fs;
 use std::os::unix::fs::MetadataExt as _;
 use std::path::{Path, PathBuf};
 use std::rc::Rc;
 use thiserror::Error;
 
+#[cfg(target_os = "linux")]
+use std::os::linux::fs::MetadataExt as _;
+#[cfg(target_os = "macos")]
+use std::os::macos::fs::MetadataExt as _;
+
 use crate::config;
 
 #[derive(Error, Debug)]
diff --git a/src/multiprocess.rs b/src/multiprocess.rs
index e1aa7ff..4ab6ce6 100644
--- a/src/multiprocess.rs
+++ b/src/multiprocess.rs
@@ -76,6 +76,7 @@ impl Controller {
             sndbuf = sys::socket::getsockopt(fd, sys::socket::sockopt::SndBuf)?;
             debug!("Tried to set socket buffer size to {}, got {}", newsize, sndbuf);
 
+            #[cfg(any(target_os = "android", target_os = "linux"))]
             if newsize > sndbuf {
                 if let Err(err) = sys::socket::setsockopt(
                     fd,
diff --git a/tests/test_handlers/test_ar.rs b/tests/test_handlers/test_ar.rs
index c1d846a..8607eb4 100644
--- a/tests/test_handlers/test_ar.rs
+++ b/tests/test_handlers/test_ar.rs
@@ -1,8 +1,11 @@
 /* SPDX-License-Identifier: GPL-3.0-or-later */
 
 use std::fs;
-use std::os::linux::fs::MetadataExt;
 use std::rc::Rc;
+#[cfg(target_os = "linux")]
+use std::os::linux::fs::MetadataExt as _;
+#[cfg(target_os = "macos")]
+use std::os::macos::fs::MetadataExt as _;
 
 use add_determinism::config;
 use add_determinism::handlers;
diff --git a/tests/test_handlers/test_javadoc.rs b/tests/test_handlers/test_javadoc.rs
index 21f865d..c5265f4 100644
--- a/tests/test_handlers/test_javadoc.rs
+++ b/tests/test_handlers/test_javadoc.rs
@@ -1,7 +1,10 @@
 /* SPDX-License-Identifier: GPL-3.0-or-later */
 
 use std::fs;
-use std::os::linux::fs::MetadataExt;
+#[cfg(target_os = "linux")]
+use std::os::linux::fs::MetadataExt as _;
+#[cfg(target_os = "macos")]
+use std::os::macos::fs::MetadataExt as _;
 
 use add_determinism::handlers;
 use add_determinism::handlers::javadoc;
diff --git a/tests/test_handlers/test_pyc.rs b/tests/test_handlers/test_pyc.rs
index d4294a3..0d014d8 100644
--- a/tests/test_handlers/test_pyc.rs
+++ b/tests/test_handlers/test_pyc.rs
@@ -3,8 +3,11 @@
 use std::io::Read;
 use std::fs;
 use std::fs::File;
-use std::os::linux::fs::MetadataExt;
 use std::path::Path;
+#[cfg(target_os = "linux")]
+use std::os::linux::fs::MetadataExt as _;
+#[cfg(target_os = "macos")]
+use std::os::macos::fs::MetadataExt as _;
 
 use add_determinism::handlers;
 use add_determinism::handlers::pyc;
diff --git a/tests/test_handlers/test_pyc_zero_mtime.rs b/tests/test_handlers/test_pyc_zero_mtime.rs
index d759199..25f9c77 100644
--- a/tests/test_handlers/test_pyc_zero_mtime.rs
+++ b/tests/test_handlers/test_pyc_zero_mtime.rs
@@ -1,8 +1,12 @@
 /* SPDX-License-Identifier: GPL-3.0-or-later */
 
 use std::fs;
-use std::os::linux::fs::MetadataExt;
 use std::time;
+#[cfg(target_os = "linux")]
+use std::os::linux::fs::MetadataExt as _;
+#[cfg(target_os = "macos")]
+use std::os::macos::fs::MetadataExt as _;
+
 
 use add_determinism::handlers;
 use add_determinism::handlers::pyc;